src/SqlPersistence.PersistenceTests/PersistenceTestsConfiguration.cs

```diff
@@ -1,174 +1,201 @@
+using NServiceBus.Persistence;
+using NServiceBus.Persistence.Sql.ScriptBuilder;
+using NServiceBus.Sagas;
+using NServiceBus.Transport;
+using NUnit.Framework;
+
+public partial class PersistenceTestsConfiguration
 {
-    using System;
-    using System.Collections.Generic;
-    using System.Threading;
-    using System.Threading.Tasks;
-    using System.Transactions;
-    using Newtonsoft.Json;
-    using NServiceBus.Extensibility;
-    using NServiceBus.Outbox;
-    using NServiceBus.Persistence;
-    using NServiceBus.Persistence.Sql.ScriptBuilder;
-    using NServiceBus.Sagas;
-    using NServiceBus.Transport;
-    using NUnit.Framework;
-
-    public partial class PersistenceTestsConfiguration
-    {
-        public bool SupportsDtc => OperatingSystem.IsWindows() && ((SqlTestVariant)Variant.Values[0]).SupportsDtc;
-
-        public bool SupportsOutbox => true;
+    public bool SupportsDtc => OperatingSystem.IsWindows() && ((SqlTestVariant)Variant.Values[0]).SupportsDtc;
 
-        public bool SupportsFinders => false;
+    public bool SupportsOutbox => true;
 
-        public bool SupportsPessimisticConcurrency { get; set; }
+    public bool SupportsFinders => false;
 
-        public ISagaIdGenerator SagaIdGenerator { get; private set; }
+    public bool SupportsPessimisticConcurrency { get; set; }
 
-        public ISagaPersister SagaStorage { get; private set; }
+    public ISagaIdGenerator SagaIdGenerator { get; private set; }
 
-        public IOutboxStorage OutboxStorage { get; private set; }
+    public ISagaPersister SagaStorage { get; private set; }
 
-        public Func<ICompletableSynchronizedStorageSession> CreateStorageSession { get; private set; }
+    public IOutboxStorage OutboxStorage { get; private set; }
 
-        static PersistenceTestsConfiguration()
-        {
-            var variants = new List<object>();
+    public Func<ICompletableSynchronizedStorageSession> CreateStorageSession { get; private set; }
 
-            if (!string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable("SQLServerConnectionString")))
-            {
-                variants.Add(CreateVariant(new SqlDialect.MsSqlServer(), BuildSqlDialect.MsSqlServer, supportsDtc: true));
-            }
+    static PersistenceTestsConfiguration()
+    {
+        var variants = new List<object>();
 
-            if (!string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable("PostgreSqlConnectionString")))
-            {
-                variants.Add(CreateVariant(new SqlDialect.PostgreSql(), BuildSqlDialect.PostgreSql));
-            }
+        var sqlServerConnectionString = Environment.GetEnvironmentVariable("SQLServerConnectionString");
+        if (!string.IsNullOrWhiteSpace(sqlServerConnectionString))
+        {
+            variants.Add(CreateVariant(new SqlDialect.MsSqlServer(),
+                BuildSqlDialect.MsSqlServer,
+                usePessimisticModeForSagas: true,
+                usePessimisticModeForOutbox: false,
```

Comment:
Tests fail when running in pessimistic mode for the outbox so this always needs to be false